### PR TITLE
FullNode: Accept some parameters as 'in'

### PR DESCRIPTION
**Pull request by Geod24**

### CODE REVIEW

**Review by kchulj — APPROVED** [review]
✅

File: source/agora/cli/client/SendTxProcess.d [hunk 1]
```diff
@@ -225,13 +226,13 @@ unittest
         /// Contains the transaction cache
         private Transaction[Hash] tx_cache;
 
-        public override void putTransaction (Transaction tx) @safe
+        public override void putTransaction (in Transaction tx) @safe
         {
-            this.tx_cache[hashFull(tx)] = tx;
+            this.tx_cache[hashFull(tx)] = tx.serializeFull().deserializeFull!Transaction;
```
Comment (linked0) [inline source/agora/cli/client/SendTxProcess.d:231]: Why do we serialize and deserialize the transaction?

Comment (kchulj) [reply source/agora/cli/client/SendTxProcess.d:231]: Good question..

Comment (Geod24) [reply source/agora/cli/client/SendTxProcess.d:231]: To deep copy it. That's the only way to get a mutable instance from a const one. And that's the only place where we need one.

**Review by linked0 — APPROVED** [review]

**Pull request merged**